staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -1445,7 +1614,9 @@ func (c *cacheWatcher) process(ctx context.Context, resourceVersion uint64) {
 			}
 			// only send events newer than resourceVersion
 			if event.ResourceVersion > resourceVersion {
-				c.sendWatchCacheEvent(event)
+				if c.sendWatchCacheEvent(event) {
```

Comment:
instead of changing the signature of sendWatchCacheEvent, let's call the c.markBookmarkAfter... directly from that only place where true is returned.

Reply:
we could do that.